### ISSUE TITLE
Missing hyperlink in generated HTML using DOT

### ISSUE THREAD

**Issue by fredArinax** [doxygen/doxygen#8055]
Hello,
It's my first collaboration...
I appears that when using DOT the generated HTML field "usemap="#" is badly formated : a "a" letter is always appended and then the name do not match below <map name and id.
So the clickable region is not working.
(see below for a sample bad formated HTML)

If not using DOT, there is no issue and the graph class are clickable. I did verify that the usemap formating is correct.

Doxygen version used: 1.8.20 (f246dd2f1c58eea39ea3f50c108019e4d4137bd5)

HTML sample (regarding a class inheritance ddiagram) with bad usemap field

```
<div class="center">
<img src="classarinax__ns_1_1arinax_exception__inherit__graph.png" border="0" usemap="#aarinax__ns_1_1arinax_exception_inherit__map" alt="Inheritance graph"/>
</div>
<map name="arinax__ns_1_1arinax_exception_inherit__map" id="arinax__ns_1_1arinax_exception_inherit__map">
```

**Comment by albert-github** [comment]
Looks like this is the same problem as reported as a regression on #7840 and fixed in the current master by means of #8012
- please check
- when not not the case:
   - Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by fredArinax** [comment]
I’m affraid it is not exactlly the same issue since in my case a « a » is ALWAYS appended what ever the filename begin with a letter, a digit or _.
But the problem is that this letter « a » is ONLY appended to the usemap="#a<myfile>"
And not on the <map name and Id fields (sorry I am not expert in HTML and certainly do not use the right terms)
<map name="<MyFile>" id="<MyFile>"

Here is a code snipset that fails:
<div class="center">
<img src="classarinax__ns_1_1arinax_exception__inherit__graph.png" border="0" usemap="#aarinax__ns_1_1arinax_exception_inherit__map" alt="Inheritance graph"/>
</div>
<map name="arinax__ns_1_1arinax_exception_inherit__map" id="arinax__ns_1_1arinax_exception_inherit__map">

The same code handy modified that works: (I just remove the 'a' letter after the #
<div class="center">
<img src="classarinax__ns_1_1arinax_exception__inherit__graph.png" border="0" usemap="#arinax__ns_1_1arinax_exception_inherit__map" alt="Inheritance graph"/>
</div>
<map name="arinax__ns_1_1arinax_exception_inherit__map" id="arinax__ns_1_1arinax_exception_inherit__map">

Hope I am clear enough...

De : albert-github [mailto:notifications@github.com]
Envoyé : jeudi 24 septembre 2020 10:01
À : doxygen/doxygen <doxygen@noreply.github.com>
Cc : Frédéric PRIN <Frederic.PRIN@arinax.com>; State change <state_change@noreply.github.com>
Objet : Re: [doxygen/doxygen] Missing hyperlink in generated HTML using DOT (#8055)

Looks like this is the same problem as reported as a regression on https://github.com/doxygen/doxygen/pull/7840 and fixed in the current master by means of https://github.com/doxygen/doxygen/pull/8012
• please check
• when not not the case:
o Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by albert-github** [comment]
Looks indeed like another case but to be able to fix it it is necessary to have an example so:
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by fredArinax** [comment]
Attached is a small example.

Navigate to file ./tmp/testDoxHTML/html/classarinax__ns_1_1_image_source.html

Look for

<img src="classarinax__ns_1_1_image_source__inherit__graph.png" border="0" usemap="#aarinax__ns_1_1_image_source_inherit__map" alt="Inheritance graph">
<map name="arinax__ns_1_1_image_source_inherit__map" id="aarinax__ns_1_1_image_source_inherit__map">


But… it’s strange since its not eactly what I saw in my full code set…
In my real case even the id="aarinax__ns_1_1_image_source_inherit__map"> field

Anyway, you can see that the first class box displaying « arinax_ns ::Source » is not clikable

Hope it helps




De : albert-github [mailto:notifications@github.com]
Envoyé : jeudi 24 septembre 2020 10:27
À : doxygen/doxygen <doxygen@noreply.github.com>
Cc : Frédéric PRIN <Frederic.PRIN@arinax.com>; State change <state_change@noreply.github.com>
Objet : Re: [doxygen/doxygen] Missing hyperlink in generated HTML using DOT (#8055)


Looks indeed like another case but to be able to fix it it is necessary to have an example so:

  *   Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by albert-github** [comment]
Sorry I don't see an attachment.

**Comment by fredArinax** [comment]
I verified that I did attached the zip… it is in my sent box.
So I retry with a « piz » extention…

Let me know if it works


De : albert-github [mailto:notifications@github.com]
Envoyé : jeudi 24 septembre 2020 10:58
À : doxygen/doxygen <doxygen@noreply.github.com>
Cc : Frédéric PRIN <Frederic.PRIN@arinax.com>; State change <state_change@noreply.github.com>
Objet : Re: [doxygen/doxygen] Missing hyperlink in generated HTML using DOT (#8055)


Sorry I don't see an attachment.

**Comment by albert-github** [comment]
Problem is that github does not accept attachments when a reply is send by email, so you have to create a comment in github and attach the file there.

**Comment by fredArinax** [comment]
here is the attached zip
[tmpIssue8055.zip](https://github.com/doxygen/doxygen/files/5274883/tmpIssue8055.zip)

**Comment by fredArinax** [comment]
done

De : albert-github [mailto:notifications@github.com]
Envoyé : jeudi 24 septembre 2020 11:06
À : doxygen/doxygen <doxygen@noreply.github.com>
Cc : Frédéric PRIN <Frederic.PRIN@arinax.com>; State change <state_change@noreply.github.com>
Objet : Re: [doxygen/doxygen] Missing hyperlink in generated HTML using DOT (#8055)


Problem is that github does not accept attachments when a reply is send by email, so you have to create a comment in github and attach the file there.

**Comment by albert-github** [comment]
In the 1.8.20 version I see (lines 81 and 82):
```
<div class="center"><img src="classarinax__ns_1_1_image_source__inherit__graph.png" border="0" usemap="#aarinax__ns_1_1_image_source_inherit__map" alt="Inheritance graph"/></div>
<map name="arinax__ns_1_1_image_source_inherit__map" id="aarinax__ns_1_1_image_source_inherit__map">
```
but in the current master I do see:
```
<div class="center"><img src="classarinax__ns_1_1_image_source__inherit__graph.png" border="0" usemap="#aarinax__ns_1_1_image_source_inherit__map" alt="Inheritance graph"/></div>
<map name="aarinax__ns_1_1_image_source_inherit__map" id="aarinax__ns_1_1_image_source_inherit__map">
```

So it looks to me that the problem is solved in the current master.

**Note:**
I saw in my browser that also in the 1.8.20 the links looked like to work, but as I was looking at another problem involving Google Chrome I also tried in it Google Chrome and here the links didn't work. With the current master the links worked for me in Firefox and in Google Chrome.
- which browser did you use?

**Comment by fredArinax** [comment]
Chrome Version 85.0.4183.102 (Build officiel) (64 bits)
, edge Version 85.0.564.51 (Version officielle) (64 bits)
and safari (don’t have the mac near me)



De : albert-github [mailto:notifications@github.com]
Envoyé : jeudi 24 septembre 2020 11:55
À : doxygen/doxygen <doxygen@noreply.github.com>
Cc : Frédéric PRIN <Frederic.PRIN@arinax.com>; State change <state_change@noreply.github.com>
Objet : Re: [doxygen/doxygen] Missing hyperlink in generated HTML using DOT (#8055)


In the 1.8.20 version I see (lines 81 and 82):

<div class="center"><img src="classarinax__ns_1_1_image_source__inherit__graph.png" border="0" usemap="#aarinax__ns_1_1_image_source_inherit__map" alt="Inheritance graph"/></div>

<map name="arinax__ns_1_1_image_source_inherit__map" id="aarinax__ns_1_1_image_source_inherit__map">

but in the current master I do see:

<div class="center"><img src="classarinax__ns_1_1_image_source__inherit__graph.png" border="0" usemap="#aarinax__ns_1_1_image_source_inherit__map" alt="Inheritance graph"/></div>

<map name="aarinax__ns_1_1_image_source_inherit__map" id="aarinax__ns_1_1_image_source_inherit__map">

So it looks to me that the problem is solved in the current master.

Note:
I saw in my browser that also in the 1.8.20 the links looked like to work, but as I was looking at another problem involving Google Chrome I also tried in it Google Chrome and here the links didn't work. With the current master the links worked for me in Firefox and in Google Chrome.

  *   which browser did you use?

**Comment by albert-github** [comment]
I just tested Microsoft Edge as well and this behaved like Google Chrome.

So I think we can say fixed but not yet released.

**Comment by fredArinax** [comment]
My issue is not linked to a browser but a bad HTML generated code regarding the map label…
And i agree that it your master code will work.
When will you about to deliver a version based on your master ?

Since it’s very long to search/replace all my projets html files !!



De : albert-github [mailto:notifications@github.com]
Envoyé : jeudi 24 septembre 2020 11:55
À : doxygen/doxygen <doxygen@noreply.github.com>
Cc : Frédéric PRIN <Frederic.PRIN@arinax.com>; State change <state_change@noreply.github.com>
Objet : Re: [doxygen/doxygen] Missing hyperlink in generated HTML using DOT (#8055)


In the 1.8.20 version I see (lines 81 and 82):

<div class="center"><img src="classarinax__ns_1_1_image_source__inherit__graph.png" border="0" usemap="#aarinax__ns_1_1_image_source_inherit__map" alt="Inheritance graph"/></div>

<map name="arinax__ns_1_1_image_source_inherit__map" id="aarinax__ns_1_1_image_source_inherit__map">

but in the current master I do see:

<div class="center"><img src="classarinax__ns_1_1_image_source__inherit__graph.png" border="0" usemap="#aarinax__ns_1_1_image_source_inherit__map" alt="Inheritance graph"/></div>

<map name="aarinax__ns_1_1_image_source_inherit__map" id="aarinax__ns_1_1_image_source_inherit__map">

So it looks to me that the problem is solved in the current master.

Note:
I saw in my browser that also in the 1.8.20 the links looked like to work, but as I was looking at another problem involving Google Chrome I also tried in it Google Chrome and here the links didn't work. With the current master the links worked for me in Firefox and in Google Chrome.

  *   which browser did you use?

**Comment by albert-github** [comment]
I have no idea when the next release will be, last release was just 1 month ago.

Best for you would probably be to:
- create an executable from master

or
- create an executable based on 1.8.20 and also incorporate the fix from #8012

**Comment by fredArinax** [comment]
And what about using a previous, released, version ?
can you tel me what is the last version when the 'a' were not appended to fix either issues #7840  #8012 ?
thanks

**Comment by albert-github** [comment]
Quick look at the dates revealed that the release before the fix in #7840 was the 1.8.18 release.
When using version 1.8.18 you will fix in most cases the 'a' problem, but there might still be cases where you have an incorrect label. It depends on how strict the browser is you use whether or not other cases kick in, the (unfortunately incomplete) change was made with a reason.
Furthermore when using the older version you will miss out on some other bug fixes / new features.

**Comment by fredArinax** [comment]
Yes I agree that downgrading is not the best solution… but I do not have time to build doxygen from the master (mainly because I did not already know about the build process and maybe I am a bit lazy !!)
Thanks a lot for all your feedback.

De : albert-github [mailto:notifications@github.com]
Envoyé : vendredi 25 septembre 2020 09:48
À : doxygen/doxygen <doxygen@noreply.github.com>
Cc : Frédéric PRIN <Frederic.PRIN@arinax.com>; State change <state_change@noreply.github.com>
Objet : Re: [doxygen/doxygen] Missing hyperlink in generated HTML using DOT (#8055)


Quick look at the dates revealed that the release before the fix in #7840<https://github.com/doxygen/doxygen/pull/7840> was the 1.8.18 release.
When using version 1.8.18 you will fix in most cases the 'a' problem, but there might still be cases where you have an incorrect label. It depends on how strict the browser is you use whether or not other cases kick in, the (unfortunately incomplete) change was made with a reason.
Furthermore when using the older version you will miss out on some other bug fixes / new features.

**Comment by doxygen** [comment]
Next release will probably not be before Christmas.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
